docs/developer/jobs.md

```diff
@@ -0,0 +1,70 @@
+# Jobs
+
```

Comment:
We should also add links here to the prow grids:

- https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api

Reply:
Done